raiden-ts/tests/unit/epics/mediate.spec.ts

```diff
@@ -0,0 +1,197 @@
+/* eslint-disable @typescript-eslint/no-explicit-any,@typescript-eslint/camelcase */
+import { bigNumberify } from 'ethers/utils';
+import { Zero, One } from 'ethers/constants';
+import { of } from 'rxjs';
+import { first } from 'rxjs/operators';
+
+import { Capabilities } from 'raiden-ts/constants';
+import { raidenConfigUpdate } from 'raiden-ts/actions';
+import { MessageType, LockedTransfer } from 'raiden-ts/messages/types';
+import { signMessage } from 'raiden-ts/messages/utils';
+import { tokenMonitored, channelOpen, channelDeposit } from 'raiden-ts/channels/actions';
+import { messageReceived } from 'raiden-ts/messages/actions';
+import { transfer } from 'raiden-ts/transfers/actions';
+import {
+  transferGenerateAndSignEnvelopeMessageEpic,
+  transferMediateEpic,
+} from 'raiden-ts/transfers/epics';
+import { UInt, Int, Signed } from 'raiden-ts/utils/types';
+import {
+  makeMessageId,
+  makeSecret,
+  getSecrethash,
+  makePaymentId,
+  getLocksroot,
+} from 'raiden-ts/transfers/utils';
+import { Direction } from 'raiden-ts/transfers/state';
+
+import { epicFixtures } from '../fixtures';
+import { raidenEpicDeps } from '../mocks';
+
+describe('mediate transfers', () => {
+  let depsMock: ReturnType<typeof raidenEpicDeps>;
+  let token: ReturnType<typeof epicFixtures>['token'],
+    tokenNetwork: ReturnType<typeof epicFixtures>['tokenNetwork'],
```

Comment:
Didn't we start splitting this into different declarations at some point?

Reply:
Not for fixtures, only for actions. This is just a way to refer to the declaration of the given function, instead of having to retype the specific types everytime; this is only needed to avoid defining these variables at declaration time, since they'll be re-set on `beforeEach`

Reply:
I would suggest starting to follow the same rules regardless of tests and production code. I don't think it makes sense to have a different approach between production and test code. This can only lead to confusion.

Reply:
I agree, test & production should follow the same rules, but specifically on this case, it's not comparable. The actions looks like simple functions, but they're redefined on each action declaration time (so much there're helpers for creating them), and there, it makes sense to have specific interfaces for the returned type (in that case, the actual action object, used in multiple places). Here, it's two one-shot functions (`raidenEpicDeps` & `epicFixtures`), which are, well, fixtures, and the second one returns a bunch of values which we destructure on the body of the suites, so these `ReturnType` are just a way to avoid re-declaring the type of these values and getting them out of sync if we change something.
It's the same (and I'm considering refactoring to it) as doing something like `let fix; fix = epicFixtures(...)`, and then access each value as `fix.value`, as it's no big deal, would just add a couple of characters before fixtures/constants usage. If you think the `ReturnType` types are bad, we could move to that, and that probably would/will be part of the #1268 issue and overall code refactoring.